server/src/main/resources/rules/rules.js

```diff
@@ -317,6 +317,21 @@ function createPool(pool) {
         return poolSet;
     };
 
+    // When pool is missing vcpu and consumer is virtual, pool should use cores
+    // instead so we calculate calculate vcpu based on cores and sockets
+    pool.calculateVcpu = function (consumer) {
+        if (Utils.isGuest(consumer) && !this.hasAttribute(VCPU_ATTRIBUTE) && this.hasAttribute(CORES_ATTRIBUTE)) {
+            pool.attributes.push({
+                name: VCPU_ATTRIBUTE,
+                value: parseInt(this.getAttribute(CORES_ATTRIBUTE))
+            });
+        }
+    };
+
+    if (consumer != null) {
+        pool.calculateVcpu(consumer);
```

Comment:
Not something I feel strongly about, but if this is the only place this method is ever getting called from, then it doesn't have to be a method. You could simply put the method's code inline in this if statement.

Reply:
Done.